### PR TITLE
Update model_deploy.

### PR DESCRIPTION
**Pull request by derekjchow**
Tested by running model_deploy_test.

### CODE REVIEW

File: slim/deployment/model_deploy.py [hunk 1]
```diff
@@ -378,8 +378,8 @@ def deploy(config,
         update_ops.append(grad_updates)
 
         update_op = tf.group(*update_ops)
-        with tf.control_dependencies([update_op]):
-          train_op = tf.identity(total_loss, name='train_op')
+        train_op = control_flow_ops.with_dependencies([update_op], total_loss,
```
Comment (sguada) [inline slim/deployment/model_deploy.py]: Rever this change, I think they want to get rid of with_dependencies

File: slim/deployment/model_deploy.py [hunk 2]
```diff
@@ -663,7 +662,7 @@ def choose(self, op):
         if op.device:
           return op.device
         node_def = op if isinstance(op, tf.NodeDef) else op.node_def
-        if node_def.op.startswith('Variable'):
+        if node_def.op in ['Variable', 'VariableV2']:
```
Comment (sguada) [inline slim/deployment/model_deploy.py]: Revert this change

File: slim/deployment/model_deploy.py (hunk 1 above)
Comment (derekjchow) [inline slim/deployment/model_deploy.py]: Done.

File: slim/deployment/model_deploy.py (hunk 2 above)
Comment (derekjchow) [inline slim/deployment/model_deploy.py]: Done.
